security/pkg/nodeagent/cache/secretcache_test.go

```diff
@@ -218,18 +213,18 @@ func TestWorkloadAgentGenerateSecretWithPluginProvider(t *testing.T) {
 }
 
 func testWorkloadAgentGenerateSecret(t *testing.T, isUsingPluginProvider bool) {
-	fakeCACli := mock.NewMockCAClient(mockCertChain1st, mockCertChainRemain, 0.1,
-		true)
+	fakeCACli, err := mock.NewMockCAClient(2, 2, time.Hour)
+	if err != nil {
+		t.Fatalf("Error creating Mock CA client: %v", err)
+	}
 	opt := Options{
-		SecretTTL:                time.Minute,
-		RotationInterval:         300 * time.Microsecond,
-		EvictionDuration:         60 * time.Second,
+		RotationInterval:         100 * time.Millisecond,
+		EvictionDuration:         300 * time.Millisecond, // Evict quickly
 		InitialBackoffInMilliSec: 10,
-		SkipValidateCert:         true,
 	}
 
 	if isUsingPluginProvider {
-		fakePlugin := mock.NewMockTokenExchangeServer()
+		fakePlugin := mock.NewMockTokenExchangeServer(3)
```

Comment:
Could you add a brief comment so that it is clear what does 3 stand for?

Reply:
Done.